tarodb/sqlc/queries/assets.sql

```diff
@@ -572,3 +572,33 @@ RETURNING script_key_id;
 SELECT script_key_id
 FROM script_keys
 WHERE tweaked_script_key = $1;
+
+-- name: UpsertAssetMeta :one
+INSERT INTO assets_meta (
+    meta_data_hash, meta_data_blob, meta_data_type
+) VALUES (
+    $1, $2, $3 
+) ON CONFLICT (meta_data_hash)
+    -- In this case, we may be inserting the data+type for an existing blob. So
+    -- we'll set both of those values. At this layer we assume the meta hash
+    -- has been validated elsewhere.
+    DO UPDATE SET meta_data_blob = EXCLUDED.meta_data_blob, 
+        meta_data_type = EXCLUDED.meta_data_type
+RETURNING meta_id;
+
+-- name: FetchAssetMeta :one
+SELECT meta_data_hash, meta_data_blob, meta_data_type
+FROM assets_meta
+WHERE meta_id = $1;
+
+-- name: FetchAssetMetaByHash :one
```

Comment:
Is the hash really unique? If I don't specify one or use the zero length string, can't there be collisions in the hashes?

Reply:
If you don't specify one, then you have the all zero meta hash. That leads to nothing (empty byte slice). 

If you use a zero length string, than that's still a meta, which leads to a different meta hash. We now require that the reveal be mandatory for genesis proofs.

Reply:
Re ppl messing with the hash, anyone can do something like: take a pdf that was time stamped, flip a single bit, then get a diff hash. For the length prefix, we use TLV so things are always canonical. So the same artifact will always produce the same hash, unless someone has messed with it.